EIPS/eip-3041.md

```diff
@@ -0,0 +1,95 @@
+---
+eip: 3041
+title: Add basefee in eth_getBlockByHash response
+author: Abdelhamid Bakhta (@abdelhamidbakhta)
+discussions-to: https://ethereum-magicians.org/t/eip-3041-add-basefee-in-eth-getblockbyhash-response/4825
+status: Draft
+type: Standards Track
+category : Interface
+created: 2020-10-13
+requires: 1559
+---
+
+## Simple Summary
+Add basefee field in `eth_getBlockByHash` RPC endpoint response.
+
+## Abstract
+[EIP-1559](/EIPS/eip-1559) introduces a base fee per gas in protocol.
+This value is maintained under consensus as a new field in the block header structure.
+This change should be reflected in RPC methods that return block header information.
+
+## Motivation
+Users may need value of the base fee at a given block. Base fee value is important to make gas price predictions more accurate.
+
+## Specification
+
+### eth_getBlockByHash
+
+#### Description
+
+Returns information about a block specified by hash.
+
+#### Parameters
+
+Parameters remain unchanged.
```

Comment:
It *may* be valuable to include the whole spec for `eth_getBlockByHash` here.  You can copy it from eip-1474 and just add in base fee.  This is optional feedback, I'm on the fence as to whether it is better to just include the change, or include the whole spec here.  When implementing from scratch, having everything in one place is very convenient.  When implementing on top of pre-existing code, just the changeset is convenient.  I'm not sure if we want to optimize for new clients or optimize for updating existing clients.

Reply:
I would personally optimize for updating existing clients. I will refer EIP-1474. If someone thinks i should put the full spec i will